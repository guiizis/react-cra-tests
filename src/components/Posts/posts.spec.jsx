import { render, screen } from '@testing-library/react';
import { Posts } from '.'

describe('<Posts>', () => {
  
  const props = [
    {
      id: '1',
      title: 'teste 1',
      body: 'teste 1',
      cover: 'img/img1.png'
    },
    {
      id: '2',
      title: 'teste 2',
      body: 'teste 2',
      cover: 'img/img2.png'
    },
    {
      id: '3',
      title: 'teste 3',
      body: 'teste 3',
      cover: 'img/img3.png'
    },
  ];

  it('should render posts', () => {
    render(<Posts posts={props}/>);

    expect(screen.getAllByRole('heading', {name: /teste/i})).toHaveLength(6);
    expect(screen.getAllByRole('img', {name: /teste/i})).toHaveLength(3);
    expect(screen.getByRole('img', {name: 'teste 3'})).toHaveAttribute('src', 'img/img3.png');
    expect(screen.getByRole('img', {name: 'teste 3'})).toHaveAttribute('src', 'img/img3.png');
  });

  it('should no render posts', () => {
    render(<Posts posts={[]}/>);

    expect(screen.queryAllByRole('heading', {name: /teste/i})).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const {container} = render(<Posts posts={props}/>);

    expect(container.firstChild).toMatchSnapshot();
  });
})