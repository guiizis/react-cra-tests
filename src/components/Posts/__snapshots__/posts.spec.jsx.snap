// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="teste 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        teste 1
      </h2>
      <h5>
        teste 1
      </h5>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="teste 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        teste 2
      </h2>
      <h5>
        teste 2
      </h5>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="teste 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        teste 3
      </h2>
      <h5>
        teste 3
      </h5>
    </div>
  </div>
</div>
`;
